### PR TITLE
[libpq] update to 12.0

### PR DESCRIPTION
**Pull request by Neumann-A**
Original #8487. Testing how far this will get on windows

### CODE REVIEW

**Comment by Neumann-A** [comment]
ah and now the interesting question is: Should we add a debug suffix to libpq or not. To d or not to d .... any suggestions?

**Comment by cenit** [comment]
To d :)

**Comment by Neumann-A** [comment]
@cenit: adding <`ImportLibrary>\$(OutDir)\$(TargetName)d.lib</ImportLibrary>` works but breaks the build of subsequent internal projects itself. Adding d in `GetAdditionalLinkerDependencies` does not work since it adds the d to all dependencies, e.g. zlibd -> zlibdd.
Naming it libpqd also goes against the maintainer guidelines. hmmm.....

Another error:
```
       Link:
         C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.23.28105\bin\HostX86\x64\link.exe /ERRORREPORT:QUEUE /OUT:".\Debug\libecpg\libecpg.dll" /INCREMENTAL:NO /NOLOGO /LIBPATH:"D:\vcpkg_test\libpq_port\installed\x64-windows\debug\lib" /LIBPATH:"D:\vcpkg_test\libpq_port\installed\x64-windows\debug\lib\manual-link" "D:/vcpkg_test/libpq_port/installed/x64-windows\debug\lib\zlibd.lib" "D:/vcpkg_test/libpq_port/installed/x64-windows\debug\lib\ssleay32.lib" "D:/vcpkg_test/libpq_port/installed/x64-windows\debug\lib\libeay32.lib" ws2_32.lib Debug/libpq/libpq.lib Debug/libpgtypes/libpgtypes.lib Debug/libpgport/libpgport.lib kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib "D:\vcpkg_test\libpq_port\installed\x64-windows\debug\lib\*.lib" /NODEFAULTLIB:libc /DEF:"src/interfaces/ecpg/ecpglib/ecpglib.def" /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /DEBUG /PDB:".\Debug\libecpg\libecpg.pdb" /SUBSYSTEM:CONSOLE /STACK:"4194304" /TLBID:1 /DYNAMICBASE:NO /NXCOMPAT /IMPLIB:".\Debug\libecpg\libecpgd.lib" /MACHINE:X64 /ignore:4197 /DLL .\Debug\libecpg\win32ver.res
         .\Debug\libecpg\connect.obj
         .\Debug\libecpg\data.obj
         .\Debug\libecpg\descriptor.obj
         .\Debug\libecpg\error.obj
         .\Debug\libecpg\execute.obj
         .\Debug\libecpg\memory.obj
         .\Debug\libecpg\misc.obj
         .\Debug\libecpg\prepare.obj
         .\Debug\libecpg\sqlda.obj
         .\Debug\libecpg\typename.obj
         "D:\vcpkg_test\libpq_port\buildtrees\libpq\x64-windows-dbg\Debug\libpq\libpqd.lib"
         "D:\vcpkg_test\libpq_port\buildtrees\libpq\x64-windows-dbg\Debug\libpgtypes\libpgtypesd.lib"
         "D:\vcpkg_test\libpq_port\buildtrees\libpq\x64-windows-dbg\Debug\libpgport\libpgport.lib"
     5>LINK : fatal error LNK1181: cannot open input file 'Debug\libpq\libpq.lib' [D:\vcpkg_test\libpq_port\buildtrees\libpq\x64-windows-dbg\libecpg.vcxproj]
```
I don't know if I want to fix that

**Comment by cenit** [comment]
Understood.
Not to d :D

File: ports/libpq/patches/windows/arm.patch [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ 	  : ($self->{type} eq "dll" ? 'DynamicLibrary' : 'StaticLibrary');
+ 	my $libs = $self->GetAdditionalLinkerDependencies($cfgname, ';');
+ 
+-	my $targetmachine =
+-	  $self->{platform} eq 'Win32' ? 'MachineX86' : 'MachineX64';
+-
++	my $targetmachine = $self->{platform} eq 'Win32' ? 'MachineX86' : 'Unknown';
++	my $targetmachine = $self->{platform} eq 'x64' ? 'MachineX64' : 'Unknown';
++	my $targetmachine = $self->{platform} eq 'ARM' ? 'MachineARM' : 'Unknown';
++	my $targetmachine = $self->{platform} eq 'ARM64' ? 'MachineARM64' : 'Unknown';
```
Comment (Neumann-A) [inline ports/libpq/patches/windows/arm.patch]: this should better be an elsif instead

**Comment by vicroms** [comment]
I must say I'm impressed by the sheer number of perl patches needed for this PR, kudos! 

It'll take me some time to review it thoroughly but with just a cursory look everything seems to be OK.

**Comment by Neumann-A** [comment]
@JackBoosY @vicroms Can somebody rerun CI for this one. The update is required so that vcpkg can move to OpenSSL 1.1

**Comment by Neumann-A** [comment]
libevent:x64-linux fails because the features are wrong .... openssl requires threading.

**Review by JackBoosY — CHANGES_REQUESTED** [review]
If there are no modifications to `vcpkg_build_make.cmake`/`vcpkg_common_functions.cmake `/`ports.cmake `, please revert them.

**Comment by Neumann-A** [comment]
@JackBoosY Reverting will be a bit difficult since your "make PR" was merged in here and fixed before it was in master. will try to copy files over from master and remove the diff that way but will wait until the qt PR (#9260) is merged

**Comment by Neumann-A** [comment]
>If there are no modifications to vcpkg_build_make.cmake/vcpkg_common_functions.cmake /ports.cmake , please revert them.

@JackBoosY: Done

File: ports/libpq/CONTROL [hunk 2]
```diff
@@ -1,5 +1,82 @@
 Source: libpq
-Version: 9.6.3
+Version: 12.0
+Build-Depends: libpq[bonjour] (osx)
```
Comment (cenit) [inline ports/libpq/CONTROL]: isn't it cleaner to have bonjour in the default features with the clause (osx)?

File: ports/x264/CONTROL [hunk 3]
```diff
@@ -1,4 +1,4 @@
 Source: x264
-Version: 157-303c484ec828ed0-6
+Version: 157-303c484ec828ed0-7
```
Comment (cenit) [inline ports/x264/CONTROL]: unnecessary bump?

File: ports/libpq/CONTROL (hunk 2 above)
Comment (Neumann-A) [reply ports/libpq/CONTROL]: Yeah but conditionals do not work in default-features. So this is a workaround for that

Comment (cenit) [reply ports/libpq/CONTROL]: ah, bummer...

File: ports/x264/CONTROL (hunk 3 above)
Comment (Neumann-A) [reply ports/x264/CONTROL]: I think this was to force a rebuild once due to a CI regression. I reverted it

**Comment by cenit** [comment]
also, another note: patches are full of "non-changes" with only the `# This must be updated` comment appended. Is it some work left to do or is it possible to clean up those patches?

**Comment by Neumann-A** [comment]
>also, another note: patches are full of "non-changes" with only the # This must be updated comment appended. Is it some work left to do or is it possible to clean up those patches?

Some of those changes in the `_release.patch` are just to keep symmetry with the `_debug.patch`. Path to GSS libs probably must be changed but the required port is not yet in VCPKG(?). Some of the OpenSSL patches could be removed though but redoing the diff without the original buildtree requires extra time.

**Pull request closed without merging**